apps/studio/components/interfaces/Organization/BillingSettings/BillingAddress/BillingAddressDialog.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from 'react'
+import { Button, Dialog, DialogContent } from 'ui'
+import { Label_Shadcn_ as Label } from 'ui'
+import BillingAddressForm from './BillingAddressForm'
+import { Pencil } from 'lucide-react'
+
+import { useCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import { useOrganizationCustomerProfileQuery } from 'data/organizations/organization-customer-profile-query'
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import NoPermission from 'components/ui/NoPermission'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import AlertError from 'components/ui/AlertError'
+
+interface BillingAddressDialogProps {
+  slug: string
+}
+
+const BillingAddressDialog = ({ slug }: BillingAddressDialogProps) => {
+  const [open, setOpen] = useState(false)
+
+  const canReadBillingAddress = useCheckPermissions(
+    PermissionAction.BILLING_READ,
+    'stripe.customer'
+  )
+
+  const canUpdateBillingAddress = useCheckPermissions(
+    PermissionAction.BILLING_WRITE,
+    'stripe.customer'
+  )
+
+  const {
```

Comment:
This wouldn't work for new organization creation, if we wanna reuse, as there is no organization yet - it would be great to have this dialog not do any actual API calls and just be functional (data-in as props, callback for saving)